R/expr-string.R

```diff
@@ -106,16 +106,21 @@ expr_str_strptime <- function(
   wrap({
     check_dots_empty0(...)
     check_polars_dtype(dtype)
-
     dtype_class <- class(dtype)
     if ("polars_dtype_datetime" %in% dtype_class) {
+      # TODO-REWRITE: why is this needed?
+      if (dtype$time_unit == "μs") {
+        dtype$time_unit <- "us"
+      }
```

Comment:
```suggestion
```

Reply:
Something like this is required, otherwise this fails:

```r
pl$DataFrame(x = "2020-01-01 01:00Z")$select(pl$col("x")$str$strptime(pl$Datetime()))

Error:
! Evaluation failed in `$select()`.
Caused by error:
! Evaluation failed in `$select()`.
Caused by error in `pl$col("x")$str$strptime()`:
! Evaluation failed in `$strptime()`.
Caused by error:
! unsupported value: 'μs'
```
If you prefer I can handle this in `impl TryFrom<&str> for Wrap<TimeUnit>` to have both "us" and "μs" return `TimeUnit::Microseconds`.

Reply:
Ah, I did not know that it returns such value!
This is a bug and will be fixed soon.